src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -63,63 +62,116 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
-                // that would defeat the purpose of this refactoring entirely).
-                var newTypeSyntax =
-                    semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType
-                                 .GenerateTypeSyntax(allowVar: false)
-                                 .WithTriviaFrom(typeSyntax);
+                var tupleTypeSymbol = semanticModel.GetTypeInfo(typeSyntax.Parent, cancellationToken).ConvertedType;
+                RoslynDebug.AssertNotNull(tupleTypeSymbol);
+
+                var leadingTrivia = declarationExpression.GetLeadingTrivia()
+                    .Concat(variableDesignation.GetAllPrecedingTriviaToPreviousToken().Where(t => !t.IsWhitespace()).Select(t => t.WithoutAnnotations(SyntaxAnnotation.ElasticAnnotation)));
 
-                Debug.Assert(!newTypeSyntax.ContainsDiagnostics, "Explicit type replacement likely introduced an error in code");
+                var tupleDeclaration = GenerateTupleDeclaration(tupleTypeSymbol, variableDesignation).WithLeadingTrivia(leadingTrivia);
 
-                editor.ReplaceNode(typeSyntax, newTypeSyntax);
+                editor.ReplaceNode(declarationExpression, tupleDeclaration);
             }
             else
             {
-                var tupleTypeSymbol = semanticModel.GetTypeInfo(typeSyntax.Parent, cancellationToken).ConvertedType;
+                var typeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
+                RoslynDebug.AssertNotNull(typeSymbol);
 
-                var leadingTrivia = node.GetLeadingTrivia()
-                    .Concat(parensDesignation.GetAllPrecedingTriviaToPreviousToken().Where(t => !t.IsWhitespace()).Select(t => t.WithoutAnnotations(SyntaxAnnotation.ElasticAnnotation)));
+                editor.ReplaceNode(typeSyntax, GenerateTypeDeclaration(typeSyntax, typeSymbol));
+            }
+        }
+
+        private static Task HandleForEachStatementAsync(Document document, SyntaxEditor editor, ForEachStatementSyntax forEach, CancellationToken cancellationToken)
+            => UpdateTypeSyntaxAsync(
+                document,
+                editor,
+                forEach.Type,
+                forEach.Identifier.Parent!,
+                cancellationToken);
+
+        private static Task HandleVariableDeclarationAsync(Document document, SyntaxEditor editor, VariableDeclarationSyntax varDecl, CancellationToken cancellationToken)
+            => UpdateTypeSyntaxAsync(
+                document,
+                editor,
+                varDecl.Type,
+                // Since we're only dealing with variable declaration using var, we know
+                // that implicitly typed variables cannot have multiple declarators in
+                // a single declaration (CS0819). Only one variable should be present
+                varDecl.Variables.Single().Identifier.Parent!,
```

Comment:
what about if it's code in error?   e.g. the user actually writes `var x, y`?

Reply:
We won't make it to this part since the fix isn't provided. Test is `TestIntLocalDeclaration_Multiple`